docs/07-lifecycle.md

```diff
@@ -2,8 +2,54 @@
+
+`APPS_PROVISION_AWS_ACCESS_KEY`
+
+`APPS_PROVISION_AWS_SECRET_KEY`
+
+We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3(not implemented yet) or from some URL(not implemented yet). Provisioning consists of 3 parts:
+
+1. Creating the lambda functions with appropriate policies
+2. Storing static assets in the dedicated S3 bucket.
+3. Storing app’s manifest file in the same dedicated S3 bucket.
+
+AWS Lambda functions have semantic names, which means that a function described in the manifest.json file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name.
+Dedicated S3 bucket name is stored in the environment variable 
+
+`MM_APPS_S3_BUCKET`
+
+which stores all apps' static assets and manifest files. 
+
+
+All files in the static folder of the bundle are considered to be the app's static assets. And are stored in the above mentioned bucket. Stored assets also have semantic keys and are generated using the rule - `static/$appID_$appVersion/filename`. 
+
+
+The `manifest.json` file of an app is stored in the same S3 bucket as well with the key - `manifests/$appID_$appVersion.json`.
+
+### Provisioning In Mattermost AWS Cloud
+To be provisioned in AWS mattermost cloud an app bundle is uploaded to the specific S3 bucket(`mattermost-apps-bucket`). On a new app release bundle is created by the circleCI and uploaded to S3(the same way plugins do right now). After that apps provisioning pipeline, running on GitlabCI, automatically detects the upload, creates appropriate lambda functions and assets the same way the **appsclt** does for the third party accounts. To generate lambda function names and asset keys apps provisioning pipeline needs additional variables for the terraform. Those are generated by the provisioning pipeline using the
+`go run ./cmd/appsctl/ generate-terraform-data /PATH/TO/YOUR/APP/BUNDLE` command or calling the `aws.GetProvisionDataFromFile(/PATH/TO/YOUR/APP/BUNDLE)` from the Proxy Plugin.
+
 ## Publish
+publishing or registering an app in a Mattermost installation means the app will be shown in the Marketplace of the installation and it can be later installed by the sysadmin and used by the users. On a totally new app registration or on a registration of the new version of the already registered app, a new version of the Proxy Plugin is cut. `apps.json` file is updated and a new app is added in the listing. Later, plugin is installed in the appropriate installations(using feature flags if necessary). After the plugin update Proxy Plugin synchronizes the list of the registered apps by downloading appropriate manifests from the S3 and storing them in memory. So the Marketplace shows renewed app listings and sysadmin can install a new app(or new version).
```

Comment:
we need to simplify terminology - published, registered, listed, (or else?)

Reply:
@iomodo ^^ is mostly for @justinegeffen to edit away, optional for this PR AFAIC.